### ISSUE TITLE
Accessibility tests are never failing using Robolectric even if there is an accessibility issue

### ISSUE THREAD

**Issue by leandrofnjr** [robolectric/robolectric#5642]
Hi, 

I've tried to implement some tests using Robolectric to catch accessibility issues, but the tests never fail even if there is an issue.

I implemented an espresso test for the same screen and one issue is always reported, but no issue is reported using Robolectric.

As you can see below, I've tried several implementations (For each attempt a different test was created) I have found from android documentation or GitHub, but I was not able to find any solution.

So, I'd like to know if there is an issue with the accessibility support for Robolectric or something wrong with my implementation.

OBS: 

I'm sorry if the Test below is not following the best standards, but I'm new with Robolectric and as it is just an investigation, I was not concerned about fix details or follow the best practices for a while 

**Robolectric version:** Latest

**_Test class using Robolectric_**
 
package com.test.roboletric

import android.os.Build
import android.view.View
import androidx.lifecycle.Lifecycle
import androidx.test.core.app.ActivityScenario
import androidx.test.espresso.Espresso
import androidx.test.espresso.action.ViewActions
import androidx.test.espresso.matcher.ViewMatchers
import androidx.test.ext.junit.runners.AndroidJUnit4
import org.junit.After
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.Robolectric
import org.robolectric.android.AccessibilityUtil
import org.robolectric.annotation.AccessibilityChecks
import org.robolectric.annotation.Config

@RunWith(AndroidJUnit4::class)
@Config(sdk = [Build.VERSION_CODES.P])
class AccessibilityTest {

    private val scenario: Lazy<ActivityScenario<MainActivity>> = lazy {
        ActivityScenario.launch(MainActivity::class.java)
    }

    @Before
    fun setUp() {
        AccessibilityUtil.setRunChecksFromRootView(true)
        AccessibilityUtil.setThrowExceptionForErrors(true)
        AccessibilityUtil.setRunChecksForRobolectricVersion(
            AccessibilityChecks.ForRobolectricVersion.LATEST
        )
    }

    @After
    fun tearDown() {
        scenario.value.close()
    }

    @Test
    @AccessibilityChecks
    fun `test using onView from Espresso library`() {
        scenario.value.moveToState(Lifecycle.State.CREATED)
        scenario.value.moveToState(Lifecycle.State.RESUMED)

        scenario.value.onActivity {
            Espresso.onView(ViewMatchers.withId(R.id.test_txt))
                .perform(ViewActions.click())
        }
    }

    @Test(expected = RuntimeException::class)
    @AccessibilityChecks
    fun `accessibility test + espresso + AccessibilityUtil checkView`() {
        scenario.value.moveToState(Lifecycle.State.CREATED)
        scenario.value.moveToState(Lifecycle.State.RESUMED)

        scenario.value.onActivity {
            val view = it.findViewById<View>(R.id.test_container)
            view.isEnabled
            view.isClickable = true
            view.importantForAccessibility = View.IMPORTANT_FOR_ACCESSIBILITY_YES

            AccessibilityUtil.checkView(view)
            AccessibilityUtil.checkView(it.findViewById(R.id.test_container))
        }
    }

    @Test
    @AccessibilityChecks
    fun `accessibility test + espresso + performClick, callOnClick, performLongClick`() {
        scenario.value.moveToState(Lifecycle.State.CREATED)
        scenario.value.moveToState(Lifecycle.State.RESUMED)

        scenario.value.onActivity {

            val view = it.findViewById<View>(R.id.test_container)
//            view.isVisible = true
            view.isEnabled
            view.isClickable = true
            view.importantForAccessibility = View.IMPORTANT_FOR_ACCESSIBILITY_YES

            view.performClick()
            view.callOnClick()
            view.performLongClick()
        }
    }

    @Test
    @AccessibilityChecks
    fun `accessibility test using Robolectric controller`() {
        val controller = Robolectric.buildActivity(MainActivity::class.java)
            .setup()
            .start()
            .resume()

        val view: View? = controller.get()?.findViewById(R.id.test_container)
        view?.isEnabled
        view?.isClickable = true
        view?.importantForAccessibility = View.IMPORTANT_FOR_ACCESSIBILITY_YES
        view?.performClick()

        // Throws java.lang.RuntimeException: Accessibility Checking requires the Android support library (v4)
        AccessibilityUtil.checkView(view)

        controller.pause()
        controller.stop()
        controller.destroy()
    }
}

**_The layout of the target screen of the test above_**

<?xml version="1.0" encoding="utf-8"?>
<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:id="@+id/test_container"
    tools:context=".MainActivity">

    <TextView
        android:id="@+id/test_txt"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:contentDescription=""
        android:text="Test Text"
        app:layout_constraintBottom_toBottomOf="parent"
        app:layout_constraintEnd_toEndOf="parent"
        app:layout_constraintStart_toStartOf="parent"
        app:layout_constraintTop_toTopOf="parent" />

    <Button
        android:id="@+id/test_bt"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:background="@android:color/white"
        android:contentDescription=""
        android:text="TEST"
        app:layout_constraintStart_toStartOf="@id/test_txt"
        app:layout_constraintTop_toBottomOf="@id/test_txt"/>

</androidx.constraintlayout.widget.ConstraintLayout>

**_The Gradle file from the module holding the test above_**

apply plugin: "com.android.application"
apply plugin: "kotlin-android"
apply plugin: "kotlin-android-extensions"

android {
    compileSdkVersion 29
    buildToolsVersion "29.0.3"

    defaultConfig {
        applicationId "com.test.roboletric"
        minSdkVersion 26
        targetSdkVersion 29
        versionCode 1
        versionName "1.0"

        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
    }

    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro"
        }
    }

    testOptions.unitTests.includeAndroidResources = true

}

dependencies {
    implementation fileTree(dir: "libs", include: ["*.jar"])
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
    implementation "androidx.appcompat:appcompat:1.1.0"
    implementation "androidx.core:core-ktx:1.2.0"
    implementation "androidx.constraintlayout:constraintlayout:1.1.3"

    testImplementation "junit:junit:4.12"
    testImplementation "org.robolectric:robolectric:4.3.1"
    testImplementation "androidx.test.espresso:espresso-core:3.2.0"
    testImplementation "androidx.test.espresso:espresso-intents:3.2.0"
    testImplementation "androidx.test.ext:junit:1.1.1"
    testImplementation "androidx.test:rules:1.2.0"
    testImplementation "androidx.test:runner:1.2.0"

    androidTestImplementation "androidx.test.ext:junit:1.1.1"
    androidTestImplementation "androidx.test.espresso:espresso-core:3.2.0"
}
`
I hope this is enough information to evaluate what is going on, if it is not, please tell which information is required for evaluation.

Thanks in advance!

**Comment by hoisie** [comment]
FWIW AccessibilityUtil is being removed from Robolectric in 4.5. It was added years ago but was never updated or supported by anyone. If you want to do accessibility checking you may have better luck using Espresso.

**Comment by leandrofnjr** [comment]
Ok, thanks for answering!

**Comment by scottyab** [comment]
Just for completeness if anyone else ends up here wondering why `AccessibilityUtil` cannot be found as hoisie mentions above it was being removed and was actually removed this [commit](https://github.com/robolectric/robolectric/commit/24c8bf78bac77d95166edeb29b145c2e0abc3b1a)
